src/frontends/common/include/openvino/frontend/frontend.hpp

```diff
@@ -20,12 +20,21 @@ namespace frontend {
 /// \brief An interface for identifying a frontend for a particular framework.
 /// Provides an ability to load and convert of input model
 class FRONTEND_API FrontEnd {
+    friend class FrontEndManager;
+    std::shared_ptr<void> m_shared_object = {};  // Library handle
```

Comment:
Please have a look https://github.com/openvinotoolkit/openvino/pull/9218 Seems the same fix is needed here

Reply:
Here we're dealing with shared_ptr's so copy pointers will not cause copy of objects. However yes, currently it is possible to copy FrontEnd objects using implicitly defined constructors. I have removed them for FrontEnd and for InputModel. For ov::Model it is not necessary as these are already deleted due to 'enable_shared_from_this'